src/graph/interface/op_def_constraint.cpp

```diff
@@ -424,6 +424,38 @@ bool check_dyn_quant_dequant_scales_zps(const op_t *n) {
     return true;
 }
 
+// check function for data_type of RMSNorm.
+bool check_rmsn_data_type(const op_t *n) {
```

Comment:
What's the difference of data type constraints between rms norm and layer norm?

Reply:
1. In oneDNN graph api doc, layernorm fp16 src/dst only support f32 gamma/beta. Primitive support both f32 and f16.  In rmns doc, align graph api with primitive. Actually, layernorm also supports f32/f16 gamma/beta for fp16 src/dst, maybe change it in another PR if needed.

layernorm doc:

| Src / Dst | Gamma / beta / mean / var|
|:----------|:-------------|
| f32       | f32          |
| bf16      | f32, bf16    |
| f16       | f32    |

rmsnorm doc:

| Src / Dst | Gamma / rrms |
|:----------|:-------------|
| f32       | f32          |
| bf16      | f32, bf16    |
| f16       | f32，f16     |

Reply:
Unless it's explicitly requested, i would suggested to keep RMSNorm and LayerNorm to have the same data type matrix. If f16 src/dst + f16 gamma/rrms are requested, probably we need to extend both ops to support that, so we will not need to maintain the subtle discrepancy.

Reply:
@yimeisun123  Is the following data type cfg required for XLA ?
src,dst: f16
gamma,rrms(optional): f16  

 If not, I'll follow Tao's comment to keep it same as layernorm.